sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -659,6 +659,16 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
     checkDataset(DatasetTransform.addOne(dataset), 2, 3, 4)
   }
 
+  test("dataset.rdd with generic case class") {
+    val ds = Seq(Generic(1, 1.0), Generic(2, 2.0)).toDS
+    val ds2 = ds.map(g => Generic(g.id, g.value))
+    ds.rdd.map(r => r.id).count
```

Comment:
Because it failed on operations of `rdd`. I will check counting results instead.